pkg/config/agentmanagement.go

```diff
@@ -114,47 +121,95 @@ func (r remoteConfigHTTPProvider) CacheRemoteConfig(remoteConfigBytes []byte) er
 	return os.WriteFile(cachePath, marshalled, 0666)
 }
 
-// FetchRemoteConfig fetches the raw bytes of the config from a remote API using
-// the values in r.AgentManagement.
-//
-// Sleeps for a short period of time to apply jitter to API requests.
-func (r remoteConfigHTTPProvider) FetchRemoteConfig() ([]byte, error) {
-	httpClientConfig := &config.HTTPClientConfig{
-		BasicAuth: &r.InitialConfig.BasicAuth,
+// readPasswordFile reads the specified password file, trims any whitespace, and returns the password as a string.
+func readPasswordFile(passwordFile string) (string, error) {
+	password, err := os.ReadFile(passwordFile)
+	if err != nil {
+		return "", errors.New("error reading password file")
 	}
+	cleanedPassword := strings.TrimSpace(string(password))
+	return cleanedPassword, nil
+}
 
-	dir, err := os.Getwd()
+// createHTTPRequest creates an HTTP request based on the values in c. If label management is enabled,
+// the request will include headers to include the agent id and indicate that label management is enabled.
+func createHTTPRequest(c *AgentManagementConfig) (*http.Request, error) {
+	// Create the full url, possibly including labels as query params
```

Comment:
I have some concerns about the code duplication and potential drift, WDYT about extending remote config and allowing to pass it headers as options (see POC [here](https://github.com/grafana/agent/commit/5f1136e1b4e7a3b3431d77b6608ef06007959bf1)). If we want to be more protocol agnostic, we could also pass it a `map[string]string` and map it to headers in the http provider.